avalon/tools/cbsceneinventory/app.py

```diff
@@ -224,6 +239,199 @@ def show_remove_warning_dialog(self, items):
+        accept_btn.setFixedWidth(24)
+        accept_btn.setFixedHeight(24)
+
+        input_layout.addWidget(accept_btn)
+
+        self._input_layout = input_layout
+        self._accept_btn = accept_btn
+
+        self.setLayout(input_layout)
+        self.setWindowTitle("Switch selected items ...")
+
+        self.connections()
+
+        self.refresh()
+
+    def connections(self):
+        self._accept_btn.clicked.connect(self._on_accept)
+
+    def refresh(self):
+
+        assets = self._get_assets()
+        self._assets_box = self._create_combo_box(assets)
+
+        subsets = self._get_subsets()
+        self._subsets_box = self._create_combo_box(subsets)
+
+        representations = self._get_representations()
+        self._representations_box = self._create_combo_box(representations)
+
+        self._input_layout.addWidget(self._representations_box)
+        self._input_layout.addWidget(self._subsets_box)
+        self._input_layout.addWidget(self._assets_box)
+
+    def _create_combo_box(self, items):
+        """Create a combobox with auto completion, first item will be "----"
+
+        Args:
+            items (iterable): list of document names
+
+        Returns:
+            QtWidgets.QComboBox
+
+        """
+
+        combobox = QtWidgets.QComboBox()
+        completer = QtWidgets.QCompleter(items)
+        combobox.setCompleter(completer)
+        combobox.setEditable(True)
+        combobox.addItem("----")
+        combobox.addItems(items)
+
+        return combobox
+
+    def _get_assets(self):
+        return self._get_documents_by_name("asset")
+
+    def _get_subsets(self):
+        return self._get_documents_by_name("subset")
+
+    def _get_representations(self):
+        return self._get_documents_by_name("representation")
+
+    def _get_documents_by_name(self, document_type, parent=None):
+
+        query = {"type": document_type}
+        if parent:
+            query["parent"] = parent["_id"]
+
+        documents = io.find(query)
+        documents_by_name = set([s["name"] for s in documents])
+
+        return sorted(list(documents_by_name))
+
+    def _get_combo_box_value(self, combo_box):
+
+        idx = combo_box.currentIndex()
+        if idx == 0:
+            return None
+        return combo_box.currentText()
+
+    def _on_accept(self):
+
+        asset = self._get_combo_box_value(self._assets_box)
+        subset = self._get_combo_box_value(self._subsets_box)
+        representation = self._get_combo_box_value(self._representations_box)
+
+        if not any([asset, subset, representation]):
+            print("Nothing selected")
+            return None
+
+        for item in self._items:
+            self.switch_item(item,
+                             asset_name=asset,
+                             subset_name=subset,
+                             representation_name=representation)
+
+        self.switched.emit()
+
+        self.close()
+
+    def switch_item(self,
```

Comment:
I always have a hard time with these kind of things by having these on the class because it's a generic method that does not use the instance of the class in any way. Always makes more sense to me more as a function instead of a method - making it look less complex.

Reply:
Maybe just push it into a `lib` for this specific tool, like how other tools have that as well (for example [project manager's lib](https://github.com/aardschok/core/blob/6c3e1a640ba78b5163a839fa57fd6ae974802cc7/avalon/tools/projectmanager/lib.py))